File: src/main/java/de/rubixdev/enchantedshulkers/ModMenuInit.java
package de.rubixdev.enchantedshulkers;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import de.rubixdev.enchantedshulkers.config.ClientConfig;
import me.shedaniel.autoconfig.AutoConfig;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;

@Environment(EnvType.CLIENT)
public class ModMenuInit implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return parent ->
                AutoConfig.getConfigScreen(ClientConfig.Inner.class, parent).get();
    }
}

File: src/main/kotlin/de/rubixdev/enchantedshulkers/ModMenuInit.kt
package de.rubixdev.enchantedshulkers

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import de.rubixdev.enchantedshulkers.config.ClientConfigImpl
import me.shedaniel.autoconfig.AutoConfig
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment

@Environment(EnvType.CLIENT)
class ModMenuInit : ModMenuApi {
    override fun getModConfigScreenFactory() = ConfigScreenFactory { parent ->
        AutoConfig.getConfigScreen(ClientConfigImpl.Inner::class.java, parent).get()
    }
}
